### PR TITLE
Update log.go requirement to v2 in go.mod and imports.

### PR DESCRIPTION
**Pull request by thomasridd**
### What
Update logging statements to format that can be parsed by
elastisearch, e.g.

log.Event(ctx, string, log.FATAL, log.Error(something1), something2)
-> log.Fatal(ctx, string, something1, something2)
log.Event(ctx, string, log.ERROR log.Error(something1), something2)
-> log.Error(ctx, string, something1, something2)
log.Event(ctx, string, log.INFO, log.Error(something))
-> log.Info(ctx, string, log.FormatErrors([]error{something}))
log.Event(ctx, string, log.WARN, log.Error(something))
-> log.Warn(ctx, string, log.FormatErrors([]error{something}))

These changes needed to allow error logs to be properly viewed in
kibana.

### How to review

Run make test
Inspect for obvious errors

### Who can review

Anyone

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -8,4 +8,8 @@ audit:
 
 build:
 	go build ./...
-.PHONY: build
\ No newline at end of file
+.PHONY: build
+
+.PHONY: lint
+lint:
+	exit
```
Comment (VivianAllen) [inline Makefile:15]: should this run `go fmt`? (see here - https://github.com/ONSdigital/dp-mongodb/blob/main/Makefile). Its inconsistent in other repos though...

File: admin.go [hunk 2]
```diff
@@ -32,11 +33,11 @@ func NewAdmin(brokerAddrs []string, pConfig *AdminConfig) (sarama.ClusterAdmin,
 
 func (t Acls) Apply(adm sarama.ClusterAdmin) error {
 	for _, acl := range t {
-		log.Event(nil, "creating ACL", log.Data{"res": acl.Resource, "acl": acl.Acl})
```
Comment (VivianAllen) [inline admin.go:35]: mybe worth a single call to context.Background() there, that can be reused on l40?

File: ci/lint.yml [hunk 3]
```diff
@@ -0,0 +1,15 @@
+---
+
+platform: linux
+
+image_resource:
+  type: docker-image
+  source:
+    repository: golang
+    tag: 1.15.7
```
Comment (VivianAllen) [inline ci/lint.yml:9]: I think 'latest' goland version here is what @nshumoogum recommended when we're adding new ones of these (a lot of the olds ymls are still old versions - there'll be another bit of work to fix that)

**Review by VivianAllen — CHANGES_REQUESTED** [review]
few tweaks I think and this is ready to go

File: ci/lint.yml (hunk 3 above)
Comment (VivianAllen) [reply ci/lint.yml:9]: NB - think I messed up the 'request changes' bit on me review - refers to this comment (everything else suggested / for discussion)

File: ci/lint.yml [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+
+inputs:
+  - name: dp-kafka
+
+run:
+  path: dp-kafka/ci/scripts/lint.sh
```
Comment (VivianAllen) [inline ci/lint.yml:15]: looks like `lint.sh` might be missing

File: Makefile (hunk 1 above)
Comment (VivianAllen) [reply Makefile:15]: following discussion with @nshumoogum this should be left as a noop

Comment (thomasridd) [reply Makefile:15]: as I understand no

File: ci/lint.yml (hunk 4 above)
Comment (thomasridd) [reply ci/lint.yml:15]: fixed that one

File: admin.go (hunk 2 above)
Comment (thomasridd) [reply admin.go:35]: unless it is a serious performance hit I'd consider it more readable having logging as a single line - depends - is there a codebase convention?

File: ci/lint.yml (hunk 3 above)
Comment (thomasridd) [reply ci/lint.yml]: done that

File: admin.go (hunk 2 above)
Comment (VivianAllen) [reply admin.go:35]: Sorry, I haven't made myself clear - I mean having ctx := context.Background() somewhere outside of the log statements, and then use the ctx variable both here and on l40? Not exactly a major issue either way

**Review by VivianAllen — APPROVED** [review]
ready to go I think

**Pull request merged**